source/MaterialXCore/Document.cpp

```diff
@@ -601,27 +601,63 @@ void Document::upgradeVersion()
         minorVersion = 36;
     }
 
-    // Upgrade path for 1.37 (change types to child outputs)
-    for (NodeDefPtr nodeDef : getNodeDefs())
+    // Upgrade path for 1.37 
+    if (majorVersion == 1 && minorVersion == 36)
```

Comment:
@feldstj Now that we have pushed the version we can use this version check.

Reply:
Does this work if the version is less than 36?

Reply:
Yes. The way upgrade works is it will bump 1 version at a time in order At line 601 we get up to version 1.36.
I actually forgot to set this at the end of this "if" so I need to fix that.